components/charity/events/CharityCard.tsx

```diff
@@ -2,15 +2,21 @@ import React from "react";
 import Image from "next/image";
 
 import { Col } from "reactstrap";
+import { CharityEvent } from "data/types";
 
-const CharityCard = ({ event, onClick }) => {
+interface CharityCardProps {
+  event: CharityEvent;
+  onClick: (event: CharityEvent) => void;
```

Comment:
Several interfaces include an `onClick` property which is eventually passed into some child component. I would highly recommend creating a generic interface in data/types.ts which these interfaces can implement. Something like

```TS
interface Clickable<T, U> {
  onClick: (input: T) => U
}
```
and you can change the interfaces to be something like
```TS
interface CharityCardProps implements Clickable<CharityEvent, void> {
  event: CharityEvent;
}
```

Reply:
You can probably do a similar thing with all the interfaces containing event but, I think that abstraction is less useful.

Reply:
Also, you can use `Clickable<void, void>` for functions that don't take any inputs. This is discouraged by TypeScript because it's seen as a 'meaningless' for an input parameter to be type 'void' but, they can't practicably stop it so you should do it.